hoot-core/src/main/cpp/hoot/core/conflate/highway/HighwayMatchCreator.cpp

```diff
@@ -213,8 +236,15 @@ class HighwayMatchVisitor : public ConstElementVisitor
     }
   }
 
-  static bool isMatchCandidate(ConstElementPtr element)
+  bool isMatchCandidate(ConstElementPtr element)
   {
+    LOG_VART(element->getElementId());
+
+    if (_filter && !_filter->isSatisfied(element))
+    {
+      return false;
+    }
+
```

Comment:
For the most part, each match creator uses an internal match visitor.  The filter logic for each one must be added like this.